Firestore/core/test/firebase/firestore/util/async_queue_test_libdispatch.cc

```diff
@@ -0,0 +1,83 @@
+/*
+ * Copyright 2018 Google
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "Firestore/core/test/firebase/firestore/util/async_queue_test.h"
+
+#include "Firestore/core/src/firebase/firestore/util/executor_libdispatch.h"
+
+#include "absl/memory/memory.h"
+#include "gtest/gtest.h"
+
+namespace firebase {
+namespace firestore {
+namespace util {
+
+namespace {
+
+dispatch_queue_t CreateDispatchQueue() {
+  return dispatch_queue_create("AsyncQueueTests", DISPATCH_QUEUE_SERIAL);
+}
+
+internal::Executor* CreateExecutorLibdispatch(const dispatch_queue_t queue) {
+  return new internal::ExecutorLibdispatch{queue};
```

Comment:
Done.

These pointers were then wrapped in `unique_ptr`s, but you're right that it would be more appropriate to return a proper type. It's a little tricky because GoogleTest's `GetParam` returns a copy of the parameter. I changed the parameter type to be a factory function instead, like in `Executor` tests.